### PR TITLE
feat:adds icons to input group

### LINKED ISSUE #536: Forms Enhancement: Add Basic Input with Icon
Add basic input with icon to https://pf-next.com/components/InputGroup/examples/

See the two calendar icon examples in the final designs here: https://drive.google.com/drive/u/1/folders/19dX04YdM2rnI00Eh-fOR4CXFt5rIq3i8

- [x] Interaction design requirements 
- [x] Visuals 
- [x] First checkpoint complete and approved 
- [ ] CSS 
- [ ] React
- [ ] Second checkpoint complete and approved

### PR DESCRIPTION
**Pull request by matthewcarleton**
This updates the input group to better use handlebars. I've also added icons. These are different from `pf-c-input-group__text` because they are styled differently and are actionable. I'm not sure if the name is the best so feedback there would be great too.
closes #536 

This also deals with some of the issues that were raised in the accessibility audit. Specifically :
#618 
#617 
#616 

Thanks!

### CODE REVIEW

File: src/patternfly/components/InputGroup/styles.scss [hunk 1]
```diff
@@ -62,6 +77,32 @@
...
+    color: var(--pf-c-input-group__action--Color);
+    cursor: pointer;
+    border: none;
+
+    &:hover{
+      color: var(--pf-c-input-group__action-hover--color);
+    }
+
+    &.pf-m-right {
+      right: 0;
```
Comment (michael-coker) [inline src/patternfly/components/InputGroup/styles.scss]: `.pf-m-align-right` and `.pf-m-align-left` would probably work better. Right now `.pf-m-top/bottom/left/right` are used to display something to the top/bottom/left/right, like a tooltip or popover or a "dropup" dropdown (uses `.pf-m-top`).

`.pf-m-alight-right/left` aligns something to the left or right edge of the parent.

File: src/patternfly/components/InputGroup/styles.scss [hunk 2]
```diff
@@ -1,31 +1,39 @@
...
+  --pf-c-input-group__text--BorderSize: var(--pf-global--BorderWidth--sm);
+  --pf-c-input-group__text--BorderColor: var(--pf-global--BorderColor--dark);
+  --pf-c-input-group__text--BackgroundColor: var(--pf-global--BackgroundColor--light-100);
+  --pf-c-input-group__action--FontSize: var(--pf-global--FontSize--md);
+  --pf-c-input-group__action--Color: var(--pf-global--Color--200);
+  --pf-c-input-group__action--PaddingTop: var(--pf-global--spacer--sm);
+  --pf-c-input-group__action--PaddingRight: var(--pf-global--spacer--sm);
+  --pf-c-input-group__action--PaddingBottom: var(--pf-global--spacer--xs);
+  --pf-c-input-group__action--PaddingLeft: var(--pf-global--spacer--sm);
+  --pf-c-input-group__action-hover--color: var(--pf-global--Color--100);
```
Comment (michael-coker) [inline src/patternfly/components/InputGroup/styles.scss]: `color` should be capitalized

File: src/patternfly/components/InputGroup/docs/code.md [hunk 3]
```diff
@@ -15,3 +15,6 @@ When using the `pf-c-input-group` always ensure labels are used outside the inpu
 | -- | -- | -- |
 | `.pf-c-input-group` | `<div>` |  Iniaties the input group **Required**. |
```
Comment (michael-coker) [inline src/patternfly/components/InputGroup/docs/code.md]: Add a period after "group"

File: src/patternfly/components/InputGroup/docs/code.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ When using the `pf-c-input-group` always ensure labels are used outside the inpu
 
 | Attribute | Applied To | Outcome |
 | -- | -- | -- |
-| `aria-describedby` | `pf-c-form-control` |  When using `pf-c-input-group__text` make use of this on the input field. |
+| `aria-describedby` | `pf-c-form-control` |  When using `pf-c-input-group__text` or `pf-c-input-group__action` make use of this on the input field. |
```
Comment (michael-coker) [inline src/patternfly/components/InputGroup/docs/code.md]: add a `.` before the classes

File: src/patternfly/components/InputGroup/styles.scss (hunk 1 above)
Comment (matthewcarleton) [reply src/patternfly/components/InputGroup/styles.scss]: 👍

**Pull request merged**